sympy/physics/mechanics/joint.py

```diff
@@ -0,0 +1,681 @@
+        3 Tuple defines the values of x, y and z directions w.r.t child's frame.
+        If it is not supplied, center of mass is added as default.
+
+    Examples
+    --------
+    Adds spherical Joint between parent's masscenter and a point at unit distance
+    along child's y axis.
+
+    >>> from sympy import Symbol
+    >>> from sympy.physics.mechanics import Body, SphericalJoint
+    >>> parent = Body('parent')
+    >>> child = Body('child')
+    >>> l = Symbol('l')
+    >>> spherical_joint = SphericalJoint('sphericaljoint', parent, child, \
+                                         child_point_pos=(0, l, 0))
+    >>> spherical_joint.coordinates
+    [sphericaljoint_alpha(t), sphericaljoint_beta(t), sphericaljoint_gamma(t)]
+    >>> spherical_joint.speeds
+    [sphericaljoint_omega_alpha(t), sphericaljoint_omega_beta(t), sphericaljoint_omega_gamma(t)]
+
+    """
+    def __init__(self, name, parent, child, parent_point_pos=None,
+                 child_point_pos=None, rot_order=None):
+
+        if rot_order is None:
+            self.rot_order = 'XYZ'
+        else:
+            self.rot_order = rot_order
+
+        super(SphericalJoint, self).__init__(name, parent, child,
+                                             parent_point_pos, child_point_pos)
+
+    def apply_joint(self):
+        alpha = dynamicsymbols(self.name + '_alpha')
+        beta = dynamicsymbols(self.name + '_beta')
+        gamma = dynamicsymbols(self.name + '_gamma')
+        alphad = dynamicsymbols(self.name + '_alpha', 1)
+        betad = dynamicsymbols(self.name + '_beta', 1)
+        gammad = dynamicsymbols(self.name + '_gamma', 1)
+        omega_alpha = dynamicsymbols(self.name + '_omega_alpha')
+        omega_beta = dynamicsymbols(self.name + '_omega_beta')
+        omega_gamma = dynamicsymbols(self.name + '_omega_gamma')
+
+        self.coordinates.append(alpha)
+        self.speeds.append(omega_alpha)
+        self.kds.append(alphad - omega_alpha)
+
+        self.coordinates.append(beta)
+        self.speeds.append(omega_beta)
+        self.kds.append(betad - omega_beta)
+
+        self.coordinates.append(gamma)
+        self.speeds.append(omega_gamma)
+        self.kds.append(gammad - omega_gamma)
+
+        self._locate_joint_point()
+        self.child_joint_point.set_pos(self.parent_joint_point, 0)
+        self.child_joint_point.set_vel(self.child.frame, 0)
+        self.child_joint_point.set_vel(self.parent.frame, 0)
+        self.parent_joint_point.set_vel(self.parent.frame, 0)
+
+        self.child.frame.orient(self.parent.frame, 'Body',
+                                [alpha, beta, gamma], self.rot_order)
+        self.child.frame.set_ang_vel(self.parent.frame,
+                                     omega_alpha * self.parent.frame.x +
+                                     omega_beta * self.parent.frame.y +
+                                     omega_gamma * self.parent.frame.z)
```

Comment:
This is incorrect. If you define the kinematical differential equations as u=q', then this is incorrect. See the three link conical pendulum example and the joints method example I created for correct definitions of these.